pontoon/machinery/views.py

```diff
@@ -69,6 +70,27 @@ def translation_memory(request):
     return JsonResponse(data, safe=False)
 
 
+def concordance_search(request):
+    """Search for translations in the internal translations memory."""
+    try:
+        text = request.GET["text"]
+        locale = request.GET["locale"]
+    except MultiValueDictKeyError as e:
+        return JsonResponse(
+            {"status": False, "message": "Bad Request: {error}".format(error=e)},
+            status=400,
+        )
+
+    try:
+        locale = Locale.objects.get(code=locale)
+    except Locale.DoesNotExist as e:
+        return JsonResponse(
+            {"status": False, "message": "Not Found: {error}".format(error=e)},
+            status=404,
+        )
+    return JsonResponse(get_concordance_search_results(text, locale), safe=False)
```

Comment:
Nitpicking: I'd make the style and naming consistent with `get_translation_memory_data` and `get_google_translate_data`:

```python

data = get_concordance_search_data(text, locale)
return JsonResponse(data, safe=False)
```

You could also rename `*_data` function names to `*_results` at the end.

Reply:
I'm going to rename the function to match the convention. I have a question about the last variable before the return statement -> do we use it to only increase the readability?

Reply:
Yeah. I don't have a strong preference for either style. But I'd prefer consistency.